std/file.d

```diff
@@ -390,6 +390,7 @@ version (Posix) private void[] readImpl(scope const(char)[] name, scope const(FS
         if (actual == 0) break;
         size += actual;
         if (size >= upTo) break;
+        assert(size <= result.length);
         if (size < result.length) continue;
```

Comment:
I agree that it is not obvious that after this line `size` needs to be equal to `result.length`, however the assert above the if makes things even more confusing. If you want to clarify things, the assert should be immediately after this if and should test for equality: `assert(size == result.length)`.